.github/workflows/wheels.yml

```diff
@@ -74,19 +74,19 @@ jobs:
         # Github Actions doesn't support pairing matrix values together, let's improvise
         # https://github.com/github/feedback/discussions/7835#discussioncomment-1769026
         buildplat:
-          - [ubuntu-20.04, manylinux_x86_64]
-          - [macos-10.15, macosx_*]
-          - [windows-2019, win_amd64]
-          - [windows-2019, win32]
+          - [ubuntu-22.04, manylinux_x86_64]
```

Comment:
These are all risky changes. The choice of windows-2019 was deliberate, also macosx. Ubuntu probably doesn't matter.
It would be best to do one thing at a time, update cibuildwheel, for instance :)

Reply:
@h-vetinari Where is condo on versioning at this point?

Reply:
> @h-vetinari Where is conda on versioning at this point?

There's a couple of things to consider. For windows, there's three related but separate things: the image, the visual studio version and the actual toolchain version. The image `windows-2019` comes with `VS2019` which _by default_  uses the vc142 toolchain. Conda-forge [moved](https://github.com/conda-forge/conda-forge.github.io/issues/1732) to this toolchain only recently. It's possible to use vc142 also from VS2022, but less obvious, and easier to get wrong.

The reason the toolchain is an issue on windows in the first place is that static libraries (like `npymath.lib`)can only be consumed by toolchains at least as new as the one used to build it, so using vc143 (the default in VS2022) would force anyone depending on that to upgrade to VS2022. For example numpy inadvertently moved to vc142 by updating away from a deprecated image, leading to https://github.com/numpy/numpy/pull/20712 & https://github.com/numpy/numpy/issues/21346, etc.

There's also efforts (by @mattip mainly) to refactor `npymath` to be a header-only lib, which would alleviate this situation.

On macos, it's possible to build for older target with newer SDKs, but the older targets (10.x) aren't supported by the SDKs >12.0 anymore AFAIR.

On linux, I don't know the relation here between `[ubuntu-22.04, manylinux_x86_64]`. Manylinux is CentOS-based, and has the lowest-denominator ABI (=glibc version etc.) that makes sure the wheels are installable on machines with an OS newer than CentOS X (depending on the manylinux version). I don't see where ubuntu enters the equation here.

In any case (mainly @EwoutH), this is a topic with potentially pretty far-reaching implications w.r.t. how much of the user base remains supported. Yes it'd be nice to live at the head, but numpy tends to sit quite far down the stack, where things move much more slowly.

Reply:
Thanks for the detailed explanation. I will remove the runner image changes from this PR, and for some images we could maybe move in a separate PR. macOS 10.15 [is deprecated](https://github.com/actions/runner-images/issues/5583) and will be removed around December 1st (within a month).

> To raise awareness of the upcoming removal, we will temporarily fail jobs using macOS 10.15. Builds that are scheduled to run during the brownout periods will fail.

Reply:
I think we have moved to the macos-11 image for testing in #22043. I don't know enough about macos to say how using this in the wheel build will affect users. Does it change the wheel tags (currently `macosx_10_9_x86_64.whl` and `macosx_11_0_arm64.whl`)?

Reply:
NumPy be like:
![bc-kwang](https://user-images.githubusercontent.com/77272/199819041-ba71f4fa-00fe-485e-87fa-a51bf7c7d401.jpeg)

Reply:
@h-vetinari Could we use MacOS 12 then? If 11 is going away we need a replacement.

Reply:
> I think we have moved to the macos-11 image for testing in #22043.

We should try that for the wheels. I wasn't worried about 1.23.x because the last release should be out before 10.15 goes away, but 1.24.x will need a later version.

Reply:
> @h-vetinari Could we use MacOS 12 then? If 11 is going away we need a replacement.

10.15 is deprecated, 11 is still fine. 

Last time I looked at this information was [here](https://github.com/conda-forge/conda-forge.github.io/issues/1798#issuecomment-1191377086), the relevant support matrix for the targets is [here](https://developer.apple.com/support/xcode/).

Of note is that even though Apple EOLs older OS versions quite aggressively, AFAIU there's still substantial numbers of users on older versions. But I don't know the Apple world so well.

Conda-forge is still at 10.9 as a default target (over 5 years EOL!), which is partially just because no-one has started a discussion on this (I think I'll open an issue...), but illustrates the point about how long the tail of this support can be.